jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -443,7 +444,14 @@ static List<HttpCookie> getCookies(Request request)
             request.getComponents().getCache().setAttribute(CACHE_ATTRIBUTE, cookieCache);
         }
 
-        cookies = cookieCache.getCookies(request.getHeaders());
+        try
+        {
+            cookies = cookieCache.getCookies(request.getHeaders());
+        }
+        catch (IllegalArgumentException iae)
```

Comment:
I think it would be less confusing and safer to introduce a `InvalidCookieException` that extends from `RuntimeException`.

Reply:
I'm generally not a huge fan of lots of exception types.... I normally ask "do you catch them and handle them differently".  In this case I am.

So I think either I change it so that any exception thrown from getCookies is caught and converted to a BadMessage(400, cause); OR I introduce an `InvalidCookieException` and catch only that.  Any other issues in the cookie parser will result in a 500 rather than a 400.